### PR TITLE
ENH: documenting supported compilers

### PR DESCRIPTION
**Pull request by dzenanz**
## PR Checklist

- [ ] :no_entry_sign: [Makes breaking changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes)
- [ ] :no_entry_sign: [Makes design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes)
- [ ] :no_entry_sign: Adds the License notice to new files.
- [ ] :no_entry_sign: Adds Python wrapping.
- [ ] :no_entry_sign: Adds tests and baseline comparison (quantitative).
- [ ] :no_entry_sign: [Adds test data](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/UploadBinaryData.md).
- [ ] :no_entry_sign: Adds Examples to [ITKExamples](https://github.com/InsightSoftwareConsortium/ITKExamples)
- [X] Adds Documentation.

### CODE REVIEW

File: Documentation/SupportedCompilers.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ITK requires a compiler with C++11 support.
+
+# Visual Studio
+* MSVC 12 (VS2013) and earlier: **NOT supported**
+* MSVC 14 (VS2015): fully supported
+* MSVC 15 (VS2017): fully supported
+* MSVC 16 (VS2019): supported with some [caveats](https://discourse.itk.org/t/has-anyone-tried-to-build-visual-studio-c-2019/1561/6?u=dzenanz)
```
Comment (blowekamp) [inline Documentation/SupportedCompilers.md]: This does not appear to be consistent with this:
https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B#Internal_version_numbering

Also please keep in mind a given Visual Studio can have multiple tool sets installed now.

**Comment by lassoan** [comment]
If we keep build instructions in the source code repository then the developer automatically gets the appropriate build instructions automatically. Browsing various versions of the same file is very easy on github web GUI.

Summary/comparison tables give a nice overview but if you need to maintain it manually (in addition to the version-specific build instructions) then it is probably not worth it. Things change too frequently and there are always some small but important details.

I agree that we should move away from the habit of referring to Visual Studio compiler version as VS20xx, as it has become ambiguous since Microsoft decoupled Visual Studio IDE and build toolset. We should mainly use v140, v141, v142 version IDs for identifying the compiler.

**Comment by lassoan** [comment]
I would consider this document to contain the official specification, so we should not require the user to look up limitations or special instructions on discourse but we should have all details here (we can still link to the discourse post for further reading).

Otherwise, this all looks good. Thank you very much for working on this!

**Comment by jhlegarreta** [comment]
It also dawned on me that users could assume that what was supported for a release version would be the version of this file pointed by the tag, so that would be easy to find out.

Let's see how things work out.

**Comment by dzenanz** [comment]
Thanks for feedback everybody! I updated Visual Studio version specifications.

@jhlegarreta A table of supported compilers for previous versions would be handy, but super hard to make. I guess we could switch to the new system of having a list of supported compilers in source code, and relying on git log/blame for history. But it will only be valid from v5 onwards.

Changing target branch from master to release, so we get this information into v5.0.1 too.

**Comment by hjmjohnson** [comment]
CI Failure on mac environment.  I think the CI Environment needs to be updated to python 3.5.

``` 
-- Found PythonLibs: /usr/lib/libpython2.7.dylib (found version "2.7.10") 
CMake Warning at Wrapping/Generators/Python/CMakeLists.txt:16 (message):
  Python executable ("2.7.16") and library ("2.7.10") version mismatch.

```

**Pull request merged**